src/cutomHooks/useUser.js

```diff
@@ -28,14 +38,14 @@ const useUser = () => {
             ? []
             : user.requests,
       };
-      dispath(populateUser(userObj));
+      dispatch(populateUser(userObj));
     } catch (error) {
       console.log(error);
     }
   };
 
   const signup = async (name, email, password, cpassword) => {
-    let url = "http://localhost:5000" + "/api/signup";
```

Comment:
ok so about this kind of hard-coded URL is bad. ideally, a project should have a .env file with a var ( named something as serverURL ) and concatenated wherever any request is sent.
This one was my bad, should have done it from the start.
@sid0-0 could you please do it with other changes